src/tally_filter.F90

```diff
@@ -0,0 +1,1510 @@
+    call write_dataset(filter_group, "bins", this % cell )
+  end subroutine to_statepoint_distribcell
+
+  subroutine to_summary_distribcell(this, filter_group)
+    class(DistribcellFilter), intent(in) :: this
+    integer(HID_T),           intent(in) :: filter_group
+
+    integer                              :: offset, k
+    character(MAX_LINE_LEN), allocatable :: paths(:)
+    character(MAX_LINE_LEN)              :: path
+
+    call write_dataset(filter_group, "type", "distribcell")
+    call write_dataset(filter_group, "n_bins", this % n_bins)
+    call write_dataset(filter_group, "bins", this % cell )
+
+    ! Write paths to reach each distribcell instance
+
+    ! Allocate array of strings for each distribcell path
+    allocate(paths(this % n_bins))
+
+    ! Store path for each distribcell instance
+    do k = 1, this % n_bins
+      path = ''
+      offset = 1
+      call find_offset(this % cell, universes(BASE_UNIVERSE), k, offset, path)
+      paths(k) = path
+    end do
+
+    ! Write array of distribcell paths to summary file
+    call write_dataset(filter_group, "paths", paths)
+    deallocate(paths)
+  end subroutine to_summary_distribcell
+
+  subroutine initialize_distribcell(this)
+    class(DistribcellFilter), intent(inout) :: this
+
+    integer :: id
+
+    ! Convert id to index.
+    id = this % cell
+    if (cell_dict % has_key(id)) then
+      this % cell = cell_dict % get_key(id)
+    else
+      call fatal_error("Could not find cell " // trim(to_str(id)) &
+           &// " specified on tally filter.")
+    end if
+  end subroutine initialize_distribcell
+
+  function text_label_distribcell(this, bin) result(label)
+    class(DistribcellFilter), intent(in) :: this
+    integer,                  intent(in) :: bin
+    character(MAX_LINE_LEN)              :: label
+
+    integer                 :: offset
+    type(Universe), pointer :: univ
+
+    univ => universes(BASE_UNIVERSE)
+    offset = 0
+    label = ''
+    call find_offset(this % cell, univ, bin-1, offset, label)
+    label = "Distributed Cell " // label
+  end function text_label_distribcell
+
+!===============================================================================
+! CellbornFilter methods
+!===============================================================================
+  subroutine get_next_bin_cellborn(this, p, estimator, current_bin, next_bin, &
+       weight)
+    class(CellbornFilter), intent(in)  :: this
+    type(Particle),        intent(in)  :: p
+    integer,               intent(in)  :: estimator
+    integer,               intent(in)  :: current_bin
+    integer,               intent(out) :: next_bin
+    real(8),               intent(out) :: weight
+
+    if (current_bin == NO_BIN_FOUND) then
+      if (this % map % has_key(p % cell_born)) then
+        next_bin = this % map % get_key(p % cell_born)
+      else
+        next_bin = NO_BIN_FOUND
+      end if
+    else
+      next_bin = NO_BIN_FOUND
+    end if
```

Comment:
Seems like this could be a bit shorter?
```Fortran
next_bin = NO_BIN_FOUND
if (current_bin == NO_BIN_FOUND) then
  if (this % map % has_key(p % cell_born)) then
    next_bin = this % map % get_key(p % cell_born)
  end if
end if
```